doc/source/train/examples/lightning/lightning_mnist_example.ipynb

```diff
@@ -0,0 +1,630 @@
+    "        loss = torch.nn.functional.cross_entropy(y_hat, y)\n",
+    "        self.log(\"train_loss\", loss)\n",
+    "        return loss\n",
+    "\n",
+    "    def validation_step(self, val_batch, batch_idx):\n",
+    "        loss, acc = self._shared_eval(val_batch)\n",
+    "        self.log(\"val_accuracy\", acc)\n",
+    "        return {\"val_loss\": loss, \"val_accuracy\": acc}\n",
+    "\n",
+    "    def test_step(self, test_batch, batch_idx):\n",
+    "        loss, acc = self._shared_eval(test_batch)\n",
+    "        self.log(\"test_accuracy\", acc)\n",
+    "        return {\"test_loss\": loss, \"test_accuracy\": acc}\n",
+    "\n",
+    "    def _shared_eval(self, batch):\n",
+    "        x, y = batch\n",
+    "        logits = self.forward(x)\n",
+    "        loss = F.nll_loss(logits, y)\n",
+    "        acc = self.accuracy(logits, y)\n",
+    "        return loss, acc\n",
+    "\n",
+    "    def validation_epoch_end(self, outputs):\n",
+    "        avg_loss = torch.stack([x[\"val_loss\"] for x in outputs]).mean()\n",
+    "        avg_acc = torch.stack([x[\"val_accuracy\"] for x in outputs]).mean()\n",
+    "        self.log(\"val_loss\", avg_loss, sync_dist=True)\n",
+    "        self.log(\"val_accuracy\", avg_acc, sync_dist=True)\n",
+    "\n",
+    "    def configure_optimizers(self):\n",
+    "        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)\n",
+    "        return optimizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You don't need to make any change to the definition of PyTorch Lightning model and datamodule."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the Cofigurations for AIR LightningTrainer\n",
+    "\n",
+    "The {meth}`LightningConfigBuilder <ray.train.lightning.LightningConfigBuilder>` class stores all the parameters involved in training a PyTorch Lightning module. It takes the same parameter lists as those in PyTorch Lightning.\n",
+    "\n",
+    "- The `.module()` method takes a subclass of `pl.LightningModule` and its initialization parameters. `LightningTrainer` will instantiate a model instance internally in the workers' training loop.\n",
+    "- The `.trainer()` method takes the initialization parameters of `pl.Trainer`. You can specify training configurations, loggers, and callbacks here.\n",
+    "- The `.fit_params()` method stores all the parameters that will be passed into `pl.Trainer.fit()`, including train/val dataloaders, datamodules, and checkpoint paths.\n",
+    "- The `.checkpointing()` method saves the configurations for a `RayModelCheckpoint` callback. This callback reports the latest metrics to the AIR session along with a newly saved checkpoint.\n",
+    "- The `.build()` method generates a dictionary that contains all the configurations in the builder. This dictionary will be passed to `LightningTrainer` later."
```

Comment:
Can you move this inline into the code cell? Colocating the text with API is easier:

like:

```
lightning_config = (
    LightningConfigBuilder()
    .module(MNISTClassifier, lr=1e-3)  # takes a subclass of `pl.LightningModule` and its initialization parameters.
    .trainer( # takes the initialization parameters of `pl.Trainer. 
        max_epochs=10,
        accelerator="cpu",
        log_every_n_steps=100,
        logger=CSVLogger("logs"),
    )
    .fit_params(datamodule=datamodule)  # stores all the parameters that will be passed into `pl.Trainer.fit()
    .checkpointing(monitor="val_accuracy", mode="max", save_top_k=3)
    .build()
)